src/com/ms/silverking/cloud/zookeeper/ZooKeeperExtended.java

```diff
@@ -42,1024 +43,1081 @@
+      return setData(path, data, version);
+    } catch (InterruptedException ie) {
+      throw new RuntimeException("panic");
     }
+  }
+
+  /**
+   * set for any version
+   *
+   * @param path
+   * @param data
+   * @return
+   * @throws KeeperException
+   */
+  public Stat set(String path, byte[] data) throws KeeperException {
+    return set(path, data, -1);
+  }
+
+  public Stat setString(String path, String data) throws KeeperException {
+    return set(path, data.getBytes());
+  }
 
-    public void deleteRecursive(String path) throws KeeperException {
-        deleteChildrenRecursive(path);
-        delete(path);
+  public Stat setInteger(String path, int x) throws KeeperException {
+    return set(path, NumConversion.intToBytes(x));
+  }
+
+  private List<ACL> getAclForPath(String path) {
+    List<ACL> aclToUse = acl.getAclForPath(path);
+    if (aclToUse == null) {
+      aclToUse = acl.getDefaultAcl();
     }
-    
-    private void deleteChildrenRecursive(String path) throws KeeperException {
-        List<String>    children;
-        List<VoidCallbackOp>   ops;
-        
-        children = getChildren(path);
-        for (String child : children) {
-            deleteChildrenRecursive(path +"/"+ child);
-        }
-        ops = new ArrayList<>(children.size());
-        for (String child : children) {
-            ops.add(deleteAsync(path +"/"+ child));
-        }
-        waitForCompletion(ops);
+    return aclToUse;
+  }
+
+  public String create(String path, byte[] data, CreateMode createMode) throws KeeperException {
+    try {
+      return create(path, data, getAclForPath(path), createMode);
+    } catch (InterruptedException ie) {
+      throw new RuntimeException("panic");
     }
+  }
+
+  public String create(String path, byte[] data) throws KeeperException {
+    return create(path, data, CreateMode.PERSISTENT);
+  }
 
-    public void createInt(String path, int val) throws KeeperException {
-        create(path, NumConversion.intToBytes(val));
+  public String create(String path) throws KeeperException {
+    return create(path, "".getBytes());
+  }
+
+  public String ensureCreated(String path) throws KeeperException {
+    if (!exists(path)) {
+      return create(path, "".getBytes());
+    } else {
+      return path;
     }
+  }
+
+  public StringCallbackOp createAsync(String path, byte[] data, List<ACL> acl, CreateMode createMode)
+      throws KeeperException {
+    StringCallbackOp op;
 
-    public int getInt(String path) throws KeeperException {
-        return getInt(path, null);
+    op = newStringCallbackOp();
+    create(path, data, acl, createMode, this, op);
+    return op;
+  }
+
+  public StringCallbackOp createAsync(String path, byte[] data) throws KeeperException {
+    return createAsync(path, data, getAclForPath(path), CreateMode.PERSISTENT);
+  }
+
+  public void delete(String path) throws KeeperException {
+    try {
+      delete(path, -1);
+    } catch (InterruptedException ie) {
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

**Problem**: InterruptedException is ignored. This can delay thread shutdown and clear the thread’s interrupt status. Only code that implements a thread’s interruption policy can swallow an interruption request.

**Fix**: Rethrow the InterruptedException or reinterrupt the current thread using *Thread.currentThread().interrupt()* so that higher-level interrupt handlers can function correctly.
 If you are wrapping the InterruptedException inside a RuntimeException, call *Thread.currentThread().interrupt()* before throwing the RuntimeException.

Learn more about [interrupts](https://docs.oracle.com/javase/tutorial/essential/concurrency/interrupt.html) and [dealing with InterruptedException](https://www.ibm.com/developerworks/java/library/j-jtp05236/index.html?ca=drs-#N10187)